### PR TITLE
[12.0][IMP] Improve usage with tips

### PR DESCRIPTION
**Pull request by foutoucour**

### CODE REVIEW

File: queue_job/readme/ROADMAP.rst [hunk 1]
```diff
@@ -16,3 +16,16 @@
 .. code-block:: sql
 
   update queue_job set state='pending' where state in ('started', 'enqueued')
+
+Tips and tricks
+---------------
+
+* **Idempotency** (https://www.restapitutorial.com/lessons/idempotency.html): The queue_job should be idempotent so they can be retried several times without impact on the data.
+* **The job should test at the very beginning its revelency**: the moment the job will be executed is unknown by design. So the first task of a job should be to check if the related work is still relevent at the moment of the execution.
```
Comment (guewen) [inline queue_job/readme/ROADMAP.rst]:
```suggestion
* **The job should test at the very beginning its relevance**: the moment the job will be executed is unknown by design. So the first task of a job should be to check if the related work is still relevant at the moment of the execution.
```

File: queue_job/readme/ROADMAP.rst [hunk 2]
```diff
@@ -16,3 +16,16 @@
...
+
+Patterns
+~~~~~~~~
+Through the time, two main patterns emerged:
+
+1. For data exposed to client, a model should store the data and the model should be the creator of the job. The job is kept hidden from the client
```
Comment (guewen) [inline queue_job/readme/ROADMAP.rst]:
```suggestion
1. For data exposed to users, a model should store the data and the model should be the creator of the job. The job is kept hidden from the users
```

File: queue_job/readme/ROADMAP.rst [hunk 3]
```diff
@@ -16,3 +16,16 @@
...
+2. For technical data, that are not exposed to the client, it is priviledged to create directly jobs, without intermediary models. The jobs are then used to check the data
```
Comment (guewen) [inline queue_job/readme/ROADMAP.rst]:
```suggestion
2. For technical data, that are not exposed to the users, it is generally alright to create directly jobs with data passed as arguments to the job, without intermediary models.
```

**Review by guewen — COMMENTED** [review]
Some typos :)

**Comment by foutoucour** [comment]
thanks !

**Review by guewen — APPROVED** [review]
Thanks

**Review by lmignon — APPROVED** [review]
Thank you @foutoucour  It's indeed useful to remember some basic principles to take care of when developing  queue jobs.

**Comment by simahawk** [comment]
NOTE: I skipped ocabot merging as squashing was needed

**Comment by sbidoul** [comment]
@simahawk if you have ideas on how the bot could support squashing, they are most welcome here: https://github.com/OCA/oca-github-bot/issues/43

**Comment by sbidoul** [comment]
Gahh. Only tilde as title underline in readme fragments... otherwise you get invalid rst. it's really something that travis should check. Currently only ocabot merge detects it... as well as the nightly wheel builder.

BTW why is this in the ROADMAP and not USAGE? @foutoucour

**Comment by simahawk** [comment]
hmm indeed, we completely missed it.  @foutoucour would you mind to check and fix?

**Pull request merged**